uplink/config_test.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"storj.io/storj/pkg/storj"
+)
+
+func TestEncryptionConfig_Key(t *testing.T) {
+	saveKey := func(key []byte) (filepath string, removeFile func()) {
+		t.Helper()
+
+		file, err := ioutil.TempFile("", "storj-test-uplink-keyfilepath-*")
+		require.NoError(t, err)
+		defer func() { require.NoError(t, file.Close()) }()
+
+		_, err = file.Write(key)
+		require.NoError(t, err)
+
+		err = file.Chmod(os.FileMode(0400))
+		require.NoError(t, err)
+
+		return file.Name(), func() { require.NoError(t, os.Remove(file.Name())) }
+	}
+
+	t.Run("ok: file with key length less or equal than max size", func(t *testing.T) {
+		someKey := make([]byte, rand.Intn(20)+1)
+		_, err := rand.Read(someKey)
+		require.NoError(t, err)
+		filename, cleanup := saveKey(someKey)
+		defer cleanup()
+
+		var expKey storj.Key
+		copy(expKey[:], someKey)
+
+		encCfg := &EncryptionConfig{
+			KeyFilepath: filename,
+		}
+		key, err := encCfg.Key()
+		require.NoError(t, err)
+
+		assert.Equal(t, expKey[:], key[:])
+	})
+
+	t.Run("ok: file with key length greater than max size", func(t *testing.T) {
+		expKey := make([]byte, rand.Intn(10)+1+storj.KeySize)
+		_, err := rand.Read(expKey)
+		require.NoError(t, err)
+		filename, cleanup := saveKey(expKey)
+		defer cleanup()
+
+		encCfg := &EncryptionConfig{
+			KeyFilepath: filename,
+		}
+		key, err := encCfg.Key()
+		require.NoError(t, err)
+
+		assert.Equal(t, expKey[:storj.KeySize], key[:])
+	})
+
+	t.Run("ok: empty file path", func(t *testing.T) {
+		filename, cleanup := saveKey([]byte{})
+		defer cleanup()
+
+		encCfg := &EncryptionConfig{
+			KeyFilepath: filename,
+		}
+		key, err := encCfg.Key()
+		require.NoError(t, err)
+		assert.Equal(t, key, storj.Key{})
+	})
+
+	t.Run("error: file not found", func(t *testing.T) {
+		// Create a temp file and delete it, to get a filepath which doesn't exist.
+		file, err := ioutil.TempFile("", "storj-test-uplink-keyfilepath-*")
```

Comment:
Why bother creating a temp file when you are going to immediately delete it?

Reply:
It's in the comment above.
And also I replied on https://github.com/storj/storj/pull/1782#discussion_r278073976